src/compiler/binder.ts

```diff
@@ -144,7 +163,10 @@ module ts {
             let symbol: Symbol;
             if (name !== undefined) {
                 symbol = hasProperty(symbols, name) ? symbols[name] : (symbols[name] = createSymbol(0, name));
-                if (symbol.flags & excludes) {
+                
+                // Check for declarations node cannot be merged with. 
+                // Interfaces declarations cannot be merged with non-ambient class declarations, which isn't encoded in SymbolFlags.

```

Comment:
I would suggest two possible options:

1. Move this check to the checker. This would involve merging interfaces and classes regardless of whether it's allowed, and just giving an error later for its correctness.
2. Add a SymbolFlag to track whether the symbol has a non-ambient class. This has the downside that it's one more thing to keep in sync, and also I think we are out of flags!

I have a strong preference for option 1.